File: fixtures/scale/kotlin/large/src/cryptobench/store/ResultRecord.kt
package cryptobench.store

import cryptobench.core.SuiteResult

data class ResultRecord(
    val suite: String,
    val passed: Int,
    val failed: Int,
    val millis: Double,
) {
    companion object {
        fun of(result: SuiteResult): ResultRecord =
            ResultRecord(result.suiteName, result.passed, result.failed, result.elapsedMillis())
    }
}

File: fixtures/scale/scala/large/src/cryptobench/store/ResultRecord.scala
package cryptobench.store

import cryptobench.core.SuiteResult

final case class ResultRecord(suite: String, passed: Int, failed: Int, millis: Double)

object ResultRecord {
  def of(result: SuiteResult): ResultRecord =
    ResultRecord(result.suiteName, result.passed, result.failed, result.elapsedMillis)
}
